db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1444,7 +1438,12 @@ void DBImpl::NotifyOnCompactionCompleted(
   TEST_SYNC_POINT("DBImpl::NotifyOnCompactionCompleted::UnlockMutex");
   {
     CompactionJobInfo info{};
-    BuildCompactionJobInfo(cfd, c, st, compaction_job_stats, job_id, current,
+    Status s = st;
+    if (s.ok() && c->is_manual_compaction() &&
```

Comment:
Hm, I checked the client code and it looks like `st` should already contain some kind of `Status::Incomplete` when it fails due to disabled manual compaction. It's also a bit late here to check since `DisableManualCompaction()` might have been called after a compaction succeeded but before this point.